interfaces/cython/cantera/base.pyx

```diff
@@ -21,25 +45,72 @@ cdef class _SolutionBase:
             self._selected_species = other._selected_species.copy()
             return
 
+        # Assign base and set managers to NULL
+        self._base = CxxNewSolution()
+        self.base = self._base.get()
+        self.thermo = NULL
+        self.kinetics = NULL
+        self.transport = NULL
+
+        # Parse inputs
         if infile.endswith('.yml') or infile.endswith('.yaml') or yaml:
-            self._init_yaml(infile, phaseid, phases, yaml)
+            self._init_yaml(infile, phase_id, adjacent, yaml)
         elif infile or source:
-            self._init_cti_xml(infile, phaseid, phases, source)
+            self._init_cti_xml(infile, phase_id, adjacent, source)
         elif thermo and species:
-            self._init_parts(thermo, species, kinetics, phases, reactions)
+            self._init_parts(thermo, species, kinetics, adjacent, reactions)
         else:
             raise ValueError("Arguments are insufficient to define a phase")
 
         # Initialization of transport is deferred to Transport.__init__
-        self.transport = NULL
+        self.base.setThermoPhase(self._thermo)
+        self.base.setKinetics(self._kinetics)
 
         self._selected_species = np.ndarray(0, dtype=np.integer)
 
     def __init__(self, *args, **kwargs):
         if isinstance(self, Transport):
             assert self.transport is not NULL
 
-    def _init_yaml(self, infile, phaseid, phases, source):
+        phase_name = pystr(self.thermo.id())
+        name = kwargs.get('name', None)
+        if name is not None:
+            self.name = name
+        elif phase_name in _phase_counts:
+            _phase_counts[phase_name] += 1
+            n = _phase_counts[phase_name]
+            self.name = '{0}_{1}'.format(phase_name, n)
+        else:
+            _phase_counts[phase_name] = 0
+            self.name = phase_name
+
+    property name:
+        """
+        The name assigned to this SolutionBase object. The default is
+        taken from the CTI/XML/YAML input file.
```

Comment:
This description isn't really accurate now, if the name is going to be replaced by a globally unique value.

Reply:
changed to: https://github.com/Cantera/cantera/blob/746c2532de768f8fb30d5f0af5490e01d8641a0b/interfaces/cython/cantera/base.pyx#L89-L91